### DISCUSSION TITLE
Detect changes made to a component's content as they are made

### DISCUSSION THREAD

**Discussion by protozoo** [GrapesJS/grapesjs#4412] [Q&A] [answered]
Hello,

I'd like to detect changes being made to a component (i.e. typing text inside a <p> element) as soon as they are made, not having to wait until the component looses focus.
I've read the issues and replies where the indicated solution is to use the "change:changesCount" event, which is the one I am already using, but this will trigger only when the component gets unselected (i.e.e click outside).
I have tried using the "component:update" event, which triggers very often, to see if I can accept the new edited content there, but does not seem to be included. 

Is there anyway to achieve what I described?

Thank you

**Comment by artf** [comment, marked as answer]
Did you try `component:input` event?

**Reply by protozoo** [reply]
I didn't. I just did and it seems to work exactly as I need, thanks!
